Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/SFTPLogger.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System.Collections.Concurrent;
+using Serilog;
+
+namespace Frends.SFTP.UploadFiles.Definitions
+{
+    /// <summary>
+    /// SFTP internal logger interface
+    /// </summary>
+    public interface ISFTPLogger : IDisposable
+    {
+        /// <summary>
+        /// Notifies of errors
+        /// </summary>
+        void NotifyError(BatchContext context, string msg, Exception e);
+
+        /// <summary>
+        /// Notifies of info-level messages
+        /// </summary>
+        void NotifyInformation(BatchContext context, string msg);
+
+        /// <summary>
+        /// Notifies of debug trace messages
+        /// </summary>
+        void NotifyTrace(string message);
+
+        /// <summary>
+        /// Logs a single successful file transfer
+        /// </summary>
+        void LogTransferSuccess(SingleFileTransfer transfer, BatchContext context);
+
+        /// <summary>
+        /// Logs a single failed file transfer
+        /// </summary>
+        void LogTransferFailed(SingleFileTransfer transfer, BatchContext context, string errorMessage, Exception exception);
+
+        /// <summary>
+        /// Logs a batch finished event
+        /// </summary>
+        void LogBatchFinished(BatchContext context, string userResultMessage, bool success, bool actionSkipped);
+    }
+
+    /// <summary>
+    /// SFTP internal logger implementation
+    /// </summary>
+    public class SFTPLogger : ISFTPLogger
```

Comment:
This can be internal class and as such most comments are not necessary as they just state the obvious